tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ConstSimplifierBase.scala

```diff
@@ -29,49 +29,170 @@ abstract class ConstSimplifierBase {
 
     // integer operations
     case OperEx(TlaArithOper.plus, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
+      ValEx(TlaInt(left + right))(intTag)
+
+    // 0 + x = x
+    case OperEx(TlaArithOper.plus, ValEx(TlaInt(left)), rightEx) =>
       if (left == 0) {
-        ValEx(TlaInt(right))(intTag)
-      } else if (right == 0) {
-        ValEx(TlaInt(left))(intTag)
+        rightEx
+      } else {
+        OperEx(TlaArithOper.plus, ValEx(TlaInt(left))(intTag), rightEx)(intTag)
+      }
+
+    // x + 0 = x
+    case OperEx(TlaArithOper.plus, leftEx, ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        leftEx
       } else {
-        ValEx(TlaInt(left + right))(intTag)
+        OperEx(TlaArithOper.plus, leftEx, ValEx(TlaInt(right))(intTag))(intTag)
       }
 
     case OperEx(TlaArithOper.minus, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
       ValEx(TlaInt(left - right))(intTag)
 
-    case ex @ OperEx(TlaArithOper.minus, NameEx(left), NameEx(right)) =>
-      if (left == right) ValEx(TlaInt(0))(intTag) else ex // this actually happens
+    // 0 - x = -x
+    case OperEx(TlaArithOper.minus, ValEx(TlaInt(left)), rightEx) =>
+      if (left == 0) {
+        OperEx(TlaArithOper.uminus, rightEx)(intTag)
+      } else {
+        OperEx(TlaArithOper.minus, ValEx(TlaInt(left))(intTag), rightEx)(intTag)
+      }
+
+    // x - 0 = x
+    case OperEx(TlaArithOper.minus, leftEx, ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        leftEx
+      } else {
+        OperEx(TlaArithOper.minus, leftEx, ValEx(TlaInt(right))(intTag))(intTag)
+      }
+
+    // x - x = 0
+    case ex @ OperEx(TlaArithOper.minus, leftEx, rightEx) =>
+      if (leftEx == rightEx) ValEx(TlaInt(0))(intTag) else ex // this actually happens
 
     case OperEx(TlaArithOper.mult, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
-      if (left == 0 || right == 0) {
-        ValEx(TlaInt(BigInt(0)))(intTag)
+      ValEx(TlaInt(left * right))(intTag)
+
+    // 0 * x = 0
+    // 1 * x = x
+    case OperEx(TlaArithOper.mult, ValEx(TlaInt(left)), rightEx) =>
+      if (left == 0) {
+        ValEx(TlaInt(0))(intTag)
       } else if (left == 1) {
-        ValEx(TlaInt(right))(intTag)
+        rightEx
+      } else {
+        OperEx(TlaArithOper.mult, ValEx(TlaInt(left))(intTag), rightEx)(intTag)
+      }
+
+    // x * 0 = 0
+    // x * 1 = x
+    case OperEx(TlaArithOper.mult, leftEx, ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        ValEx(TlaInt(0))(intTag)
+      } else if (right == 1) {
+        leftEx
+      } else {
+        OperEx(TlaArithOper.mult, leftEx, ValEx(TlaInt(right))(intTag))(intTag)
+      }
+
+    case ex @ OperEx(TlaArithOper.div, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        throw new IllegalArgumentException(s"Division by zero at ${ex.toString}")
```

Comment:
I think a custom exception might be more fitting here. iiuc, Java's [`IlegalArugemntException`](Thrown to indicate that a method has been passed an illegal or inappropriate argument.) is meant to be

> Thrown to indicate that a method has been passed an illegal or inappropriate argument.

But I don't think that's really what's happening here. It's more like an input error, iiuc. 

Perhaps this should raise a `TalInputError`? https://github.com/informalsystems/apalache/blob/406fcf0e3b46b4c6abde49c57b2981eea3a43c4a/tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/exceptions.scala#L29-L37